src/generator/models.ts

```diff
@@ -158,10 +158,12 @@ class StructDef {
         readOnly = ` azure:"ro"`;
       }
       let tag = ` \`${this.Language.marshallingFormat}:"${serialization}"${readOnly}\``;
-      // if this is a response type then omit the tag IFF the marshalling format is
-      // JSON, it's a header or is the RawResponse field.  XML marshalling needs a tag.
+      // if this is a response type then omit the tag IFF the marshalling formatvalue is non-scalar
+      // for JSON marshalling types or it's a header.  XML marshalling needs a tag.
       // also omit the tag for additionalProperties
-      if ((this.Language.responseType === true && (this.Language.marshallingFormat !== 'xml' || prop.language.go!.name === 'RawResponse')) || prop.language.go!.isAdditionalProperties) {
+      if (prop.language.go!.name === 'RawResponse' && prop.language.go!.addTag === true && this.Language.marshallingFormat !== 'xml') {
```

Comment:
I think this can be simplified to remove the need for `addTag`.
`if (prop.language.go!.name === 'RawResponse' && this.Language.marshallingFormat !== 'xml')`

Reply:
So I need to leave `addTag` as part of the check here, because "RawResponse" can exist both on scalar and non-scalar response envelopes and I think it's the cleanest way of making sure it's only added on RawResponse when the other field is a scalar. 
Add tag is set during the transform when creating the scalar type response envelope, there for no additional looping or processing is needed later on.